src/main/java/dev/morling/onebrc/CalculateAverage_gonix.java

```diff
@@ -205,27 +205,15 @@ public Stream<Entry> stream() {
 
     private void add(ByteBuffer buf, int start, long tailAndLen, int hash, int measurement) {
         int idx = hash & INDEX_MASK;
-        while (true) {
-            if (index[idx] != 0) {
-                int offset = index[idx];
-                if (keyEqual(offset, buf, start, tailAndLen)) {
```

Comment:
How do you deal with key collisions now?

Reply:
The same as before: by comparing the key. I've moved entry update logic into the `keyEqual()` method and renamed it to `update()`, which returns `false` if the keys are not equal (collision), in which case the search continues to the next slot.

Reply:
Ah yes, got it. Thx!